app/scripts/components/exploration/components/dataset-selector-modal/index.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { useAtom } from 'jotai';
+
+import { DatasetData, DatasetLayer } from 'veda';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader
+} from '@devseed-ui/modal';
+import { glsp, themeVal } from '@devseed-ui/theme-provider';
+
+
+import { timelineDatasetsAtom } from '../../atoms/datasets';
+import {
+  allDatasets as rawAllDatasets,
+  reconcileDatasets
+} from '../../data-utils';
+import RenderModalHeader from './header';
+import ModalContentRender from './content';
+import ModalFooterRender from './footer';
+
+import {
+  Actions,
+  useBrowserControls
+} from '$components/common/browse-controls/use-browse-controls';
+import { prepareDatasets, sortOptions } from '$components/data-catalog';
+
+
+const DatasetModal = styled(Modal)`
+  z-index: ${themeVal('zIndices.modal')};
+
+  /* Override ModalContents */
+  > div {
+    display: flex;
+    flex-flow: column;
+  }
+
+  ${ModalHeader} {
+    position: sticky;
+    top: ${glsp(-2)};
+    z-index: 100;
+    background-color: ${themeVal('color.base-50')};
+    align-items: start;
+    padding-top: ${glsp(2)};
+    padding-bottom: ${glsp(1)};
+    box-shadow: 0 -1px 0 0 ${themeVal('color.base-200a')};
+  }
+
+  ${ModalBody} {
+    max-height: 100%;
+    height: 400px;
+    overflow-y: auto;
+    display: flex;
+    flex-flow: column;
+    padding-top: ${glsp(2)};
+    gap: ${glsp(1)};
+  }
+
+  ${ModalFooter} {
+    display: flex;
+    gap: ${glsp(1)};
+    align-items: center;
+    position: sticky;
+    bottom: ${glsp(-2)};
+    z-index: 100;
+    padding-top: ${glsp(1)};
+    padding-bottom: ${glsp(1)};
+    background-color: ${themeVal('color.base-50')};
+    box-shadow: 0 -1px 0 0 ${themeVal('color.base-200a')};
+    > .selection-info {
+      margin-right: auto;
+    }
+  }
+`;
+
+interface DatasetSelectorModalProps {
+  revealed: boolean;
+  close: () => void;
+}
+
+const allDatasets = rawAllDatasets.map(currentDataset => {
+  return {
+    ...currentDataset,
+    layers: currentDataset.layers.map(l => ({
+      ...l,
+      parentDataset: {
+        id: currentDataset.id,
+        name: currentDataset.name
+        // infoDescription: currentDataset.infoDescription,
```

Comment:
@sandrahoang686 This is where I set up parentDataset available to layer. 
I thought about putting all the needed information on here (like infoDescription HTML and taxonomies) but decided to keep it minimal since this is enough information for a layer to get the parent data it needs. (Like how data card gets taxonomy data using this info https://github.com/NASA-IMPACT/veda-ui/pull/845/files#diff-a989668bae4be591a5f4ab7f357c63286837ce1702280c2bf988f72c758e5cb6R180)  How do you think?

Reply:
💭 : That makes sense, I think this minimal approach is the better approach to go 👍🏼 . I'm thinking though that it might be better to move this method in `../data-utils` ? And we should type the return value as well even if its just used once or twice including the data-layer-cards.